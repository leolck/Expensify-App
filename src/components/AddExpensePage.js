import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { start_add_expense } from '../actions/expenses';


export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.start_add_expense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        ); 
    } 
}

const mapDispatchToProps = (dispatch) => ({
    start_add_expense: (expense) => dispatch(start_add_expense(expense))
});



export default connect(undefined, mapDispatchToProps)(AddExpensePage);