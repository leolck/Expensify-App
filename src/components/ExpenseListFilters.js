import React from 'react';
import { connect } from 'react-redux';
import { set_text_filter, sort_by_date, sort_by_amount, set_start_date, set_end_date } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({ startDate, endDate }) => {
        this.props.set_start_date(startDate);
        this.props.set_end_date(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused: calendarFocused }))
    };
    onTextChange = (e) => {
        this.props.set_text_filter(e.target.value);
    };
    onSortChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sort_by_date();
        } else if (e.target.value === 'amount') {
            this.props.sort_by_amount();
        }
    };
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            type='text' 
                            className="text-input"
                            placeholder="Search expenses"
                            value={this.props.filters.text} 
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select 
                            className="select"
                            value={this.props.filters.sortBy} 
                            onChange={this.onSortChange}
                            >
                            <option value='date'>Date</option>
                            <option value='amount'>Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            showClearDates={true}
                        />
                    </div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    set_text_filter: (text) => dispatch(set_text_filter(text)),
    sort_by_date: () => dispatch(sort_by_date),
    sort_by_amount: () => dispatch(sort_by_amount),
    set_start_date: (startDate) => dispatch(set_start_date(startDate)),
    set_end_date: (endDate) => dispatch(set_end_date(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)