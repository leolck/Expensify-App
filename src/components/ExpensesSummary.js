import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenses }) => {
    const expenseWordHidden = hiddenExpenses === 1 ? 'expense' : 'expenses';
    const noFilters = hiddenExpenses === 0;
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span>
                </h1>
                <h2 className="page-header__subtitle">
                    {
                        noFilters
                        ?
                        <i>Showing all expenses. No filters in place.</i>
                        :
                        <i><span>{hiddenExpenses}</span> {expenseWordHidden} hidden because of your filters.</i>
                    }
                </h2>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        hiddenExpenses: state.expenses.length - visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);

