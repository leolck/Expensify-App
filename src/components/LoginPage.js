import React from 'react';
import { connect } from 'react-redux';
import { startLoginFacebook, startLoginGithub, startLoginGoogle, startLoginTwitter } from '../actions/auth';

// Fonts (Font Awesome)
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const LoginPage = ({ startLoginFacebook, startLoginGithub, startLoginGoogle, startLoginTwitter }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <div className="box-layout__body">
                <div className="box-layout__body-login">
                    <button     
                        className="button-login button--facebook"
                        onClick={startLoginFacebook}
                    >
                        <FontAwesomeIcon icon={["fab", "facebook-f"]} pull="left"></FontAwesomeIcon>
                        Login with Facebook
                    </button>
                </div>
                <div className="box-layout__body-login">
                    <button
                        className="button-login button--github"
                        onClick={startLoginGithub}
                    >
                        <FontAwesomeIcon icon={["fab", "github"]} pull="left"></FontAwesomeIcon>
                        Login with Github
                    </button>
                </div>
                <div className="box-layout__body-login">
                    <button 
                        className="button-login button--google" 
                        onClick={startLoginGoogle}
                    >
                        <FontAwesomeIcon icon={["fab", "google"]} pull="left"></FontAwesomeIcon>
                        Login with Google
                    </button>
                </div>
                <div className="box-layout__body-login">
                    <button
                        className="button-login button--twitter"
                        onClick={startLoginTwitter}
                    >
                        <FontAwesomeIcon icon={["fab", "twitter"]} pull="left"></FontAwesomeIcon>
                        Login with Twitter
                    </button>
                </div>
            </div>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch)  => ({
    startLoginFacebook: () => dispatch(startLoginFacebook()),
    startLoginGithub: () => dispatch(startLoginGithub()),
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginTwitter: () => dispatch(startLoginTwitter())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);



