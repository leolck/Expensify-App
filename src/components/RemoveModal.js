import React from 'react';
import Modal from 'react-modal';

const RemoveModal = ({ confirmRemove, onConfirmRemove, onDenyRemove }) => {
    return (
        <Modal
            isOpen={confirmRemove}
            onRequestClose={onDenyRemove}
            contentLabel='confirmation'
            closeTimeoutMS={200}
            className="modal"
        >
            <h3 className="modal__title">Are you sure?</h3>
            <div className="modal__buttons">
                <button 
                    className="button button--confirm"
                    onClick={onConfirmRemove}
                > 
                    Yes
                </button>
                <button
                    className="button button--deny"
                    onClick={onDenyRemove}
                >
                    No
                </button>
            </div>
        </Modal>
    );
};

export default RemoveModal;


// function OptionModal(props) {
//     return (
//         <Modal
//             isOpen={!!props.selectedOption}
//             onRequestClose={props.handleClearSelectedOption}
//             contentLabel='Selected Option'
//             closeTimeoutMS={200}
//             className="modal"
//         >
//             <h3 className='modal__title'>Selected Option</h3>
//             {props.selectedOption && <p className='modal__body'>{props.selectedOption}</p>}
//             <button className='button' onClick={props.handleClearSelectedOption}>Okay</button>   
//         </Modal>
//     );
// };