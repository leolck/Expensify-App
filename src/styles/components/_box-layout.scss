.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, .30);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 45rem;
    height: 30rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}

.box-layout__body {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-around;
}

.box-layout__body-login {
    width: 20rem;
    // display: flex;
    // flex-direction: row;
    // justify-content: space-between;
}

button > span {
    padding: $s-size;
}