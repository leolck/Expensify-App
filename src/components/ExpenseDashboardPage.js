import React from 'react';
import ExpensesSummary from './ExpensesSummary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

export const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;