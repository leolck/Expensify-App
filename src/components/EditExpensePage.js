import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { start_edit_expense, start_remove_expense } from '../actions/expenses';
import RemoveModal from './RemoveModal';

export class EditExpensePage extends React.Component {
    state = {
        confirmRemove: false
    };
    onSubmit = (expense) => {
        this.props.start_edit_expense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onClick = () => {
        this.setState((prevState) => ({
            confirmRemove: true
        }));
    };
    onConfirmRemove = () => {
        this.setState(() => ({
            confirmRemove: false
        }));
        this.props.start_remove_expense({ id: this.props.expense.id });
        this.props.history.push('/');
    };
    onDenyRemove = () => {
        this.setState(() => ({
            confirmRemove: false
        }));
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}  
                    />
                    <button 
                        className="button button--secondary"
                        onClick={this.onClick}
                    >
                        Remove Expense
                    </button>
                </div>
                <RemoveModal 
                    confirmRemove={this.state.confirmRemove}
                    onConfirmRemove={this.onConfirmRemove}
                    onDenyRemove={this.onDenyRemove}
                />
            </div>
        );
    }
};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
    start_edit_expense: (id, expense) => dispatch(start_edit_expense(id, expense)),
    start_remove_expense: (data) => dispatch(start_remove_expense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);