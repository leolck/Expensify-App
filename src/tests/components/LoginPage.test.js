import React from 'react';
import { LoginPage } from '../../components/LoginPage';
import { shallow } from 'enzyme';

let startLoginFacebook, startLoginGithub, startLoginGoogle, startLoginTwitter, wrapper;

beforeEach(() => {
    startLoginFacebook = jest.fn();
    startLoginGithub = jest.fn();
    startLoginGoogle = jest.fn();
    startLoginTwitter = jest.fn();
    wrapper = shallow(
        <LoginPage 
            startLoginFacebook={startLoginFacebook}
            startLoginGithub={startLoginGithub}
            startLoginGoogle={startLoginGoogle}
            startLoginTwitter={startLoginTwitter}
        />
    );
});

test('should render LoginPage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLoginFacebook on button click', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(startLoginFacebook).toHaveBeenCalled();
});

test('should call startLoginGithub on button click', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(startLoginGithub).toHaveBeenCalled();
});

test('should call startLoginGoogle on button click', () => {
    wrapper.find('button').at(2).simulate('click');
    expect(startLoginGoogle).toHaveBeenCalled();
});

test('should call startLoginTwitter on button click', () => {
    wrapper.find('button').at(3).simulate('click');
    expect(startLoginTwitter).toHaveBeenCalled();
});