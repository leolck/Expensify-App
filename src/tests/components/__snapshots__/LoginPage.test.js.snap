// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <div
      className="box-layout__body"
    >
      <div
        className="box-layout__body-login"
      >
        <button
          className="button-login button--facebook"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fab",
                "facebook-f",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull="left"
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          Login with Facebook
        </button>
      </div>
      <div
        className="box-layout__body-login"
      >
        <button
          className="button-login button--github"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fab",
                "github",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull="left"
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          Login with Github
        </button>
      </div>
      <div
        className="box-layout__body-login"
      >
        <button
          className="button-login button--google"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fab",
                "google",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull="left"
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          Login with Google
        </button>
      </div>
      <div
        className="box-layout__body-login"
      >
        <button
          className="button-login button--twitter"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "fab",
                "twitter",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull="left"
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          Login with Twitter
        </button>
      </div>
    </div>
  </div>
</div>
`;
